Core/Core/Helpers/State.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+
+namespace Speckle.Core.Helpers;
+
+public class State<T> : IDisposable where T : State<T>, new()
+{
+  static T root;
```

Comment:
I'd love to get more info on why `ThreadStatic` wasn't working for your use-case.

Reply:
I removed the `ThreadStatic` and use static variables here, because the conversion runs on multiple threads thanks to `async` functions. With `ThreadStatic` the first Peek call in a specific thread would return a new context.

Reply:
Ok this sounds legit, I suspected as much but I hadn't actually done much testing in this regard 🙌🏼

I'm ok to merge this in 🚀